src/Microsoft.Data/BaseColumn.cs

```diff
@@ -64,24 +67,26 @@ protected set
         /// <returns></returns>
         public virtual BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false) => throw new NotImplementedException();
 
+        public virtual BaseColumn CloneAndAppendNulls(long numberOfNulls, BaseColumn mapIndices = null, bool invertMapIndices = false) => throw new NotImplementedException();
+
         public virtual BaseColumn Sort(bool ascending = true) => throw new NotImplementedException();
 
         internal virtual BaseColumn GetAscendingSortIndices() => throw new NotImplementedException();
 
         internal delegate long GetBufferSortIndex(int bufferIndex, int sortIndex);
-        internal delegate T GetValueAtBuffer<T>(int bufferIndex, int valueIndex);
+        internal delegate ValueTuple<T, int> GetValueAndBufferSortIndexAtBuffer<T>(int bufferIndex, int valueIndex);
         internal delegate int GetBufferLengthAtIndex(int bufferIndex);
-        internal void PopulateColumnSortIndicesWithHeap<T>(SortedDictionary<T, List<Tuple<int, int>>> heapOfValueAndListOfTupleOfSortAndBufferIndex,
+        internal void PopulateColumnSortIndicesWithHeap<T>(SortedDictionary<T, List<ValueTuple<int, int>>> heapOfValueAndListOfTupleOfSortAndBufferIndex,
                                                             PrimitiveColumn<long> columnSortIndices,
                                                             GetBufferSortIndex getBufferSortIndex,
-                                                            GetValueAtBuffer<T> getValueAtBuffer,
+                                                            GetValueAndBufferSortIndexAtBuffer<T> getValueAndBufferSortIndexAtBuffer,
                                                             GetBufferLengthAtIndex getBufferLengthAtIndex)
         {
             while (heapOfValueAndListOfTupleOfSortAndBufferIndex.Count > 0)
             {
-                KeyValuePair<T, List<Tuple<int, int>>> minElement = heapOfValueAndListOfTupleOfSortAndBufferIndex.ElementAt(0);
-                List<Tuple<int, int>> tuplesOfSortAndBufferIndex = minElement.Value;
-                Tuple<int, int> sortAndBufferIndex;
+                KeyValuePair<T, List<ValueTuple<int, int>>> minElement = heapOfValueAndListOfTupleOfSortAndBufferIndex.ElementAt(0);
```

Comment:
```suggestion
                KeyValuePair<T, List<(int, int)>> minElement = heapOfValueAndListOfTupleOfSortAndBufferIndex.ElementAt(0);
```
Seems more readable to me. Other places as well.